### PR TITLE
Additional properties: false to allow better type checking from JSON schemas

### LINKED ISSUE #3382: JSON Schemas generated by API Platform should include additionalProperties: false
**API Platform version(s) affected**: 2.5.1

**Description**  
Today, JSON Schemas generated by API-Platform do not include `additionalProperties: false`. This reduces type saftey when using a code generator.

**How to reproduce**  
Create an ApiResource in a new project, check the generated Swagger documentation. The JSON Schema generated for the ApiResource does not include `additionalProperties: false`.

**Possible Solution**  
Simply add `additionalProperties: false` when generating the schema.

### PR DESCRIPTION
**Pull request by quentinus95**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tickets       | fixes #3382 
| License       | MIT
| Doc PR        | -

This change sets additional properties to false on all definitions so developers using code generators (e.g. TypeScript code generators from JSON schema) can have safer type checks.

### CODE REVIEW

**Comment by alanpoulain** [comment]
You should fix the CS too :slightly_smiling_face:

**Comment by quentinus95** [comment]
Done! I was double checking the failing tests as well but it seems unrelated.

**Comment by teohhanhui** [comment]
Let's wait for https://github.com/api-platform/core/pull/3402 first, as it'll make use of `additionalProperties`.

**Comment by quentinus95** [comment]
@teohhanhui is there anything I can do to speed up the merging process? Having this change merged and released would highly reduce the bug risks in my production-deployed TypeScript consumers.

**Comment by soyuka** [comment]
@quentinus95 we need to progress at our own pace :|. You should be able to use your fork (and this patch) by tweaking the `composer.json` file on your project. Would that work temporarily?

**Comment by quentinus95** [comment]
@soyuka sure! It was just to be 100% sure nothing more was required on my side.

**Comment by quentinus95** [comment]
@teohhanhui any feedback now that #3402 has been merged?

**Comment by soyuka** [comment]
could you rebase? thanks!

File: src/JsonSchema/SchemaFactory.php [hunk 1]
```diff
@@ -110,6 +110,7 @@ public function buildSchema(string $className, string $format = 'json', string $
         }
 
         $definition = new \ArrayObject(['type' => 'object']);
+        $definition['additionalProperties'] = false;
```
Comment (dunglas) [inline src/JsonSchema/SchemaFactory.php:113]: We should set this flag only if the `allow_extra_attributes` normalization context option is set to `false`: https://github.com/api-platform/core/issues/1217

**Comment by soyuka** [comment]
Thanks @quentinus95 ! Superseeded by https://github.com/api-platform/core/pull/3804

**Pull request closed without merging**